ci/scripts/build_all_examples.sh

```diff
@@ -39,7 +48,18 @@ function build_all_configs() {
                 continue
             fi
             
-            build_example "$example" "$sdk_version" "$board" "$sd_variant" "$toolchain" "$config_dir" "$build_dir" "$log_level" || exit 1
+            local start_ts=`date +%s%N`
+
+            local build_status
+            build_example "$example" "$sdk_version" "$board" "$sd_variant" "$toolchain" "$config_dir" "$build_dir" "$log_level"
+            [[ $? -eq 0 ]] && build_status="success" || build_status="failure" 
+
+            local end_ts=`date +%s%N`
+            local build_time_ms=$(ns_to_ms `expr $end_ts - $start_ts`)
+            
+            if [[ -f $build_summary_file ]]; then
+                printf "$BUILD_SUMMARY_ENTRY_FORMAT" "$example" "$board" "$sd_variant" "$sdk_version" "$build_time_ms ms" "$build_status" >> "$build_summary_file"
+            fi
```

Comment:
Shouldn't we exit with code 1 to allow CI to recognize a failure?

Reply:
I'd make it an option. I found it helpful to be able to create a report in a single run that tells me which examples build correctly and which do not.

Reply:
Yes, I was talking only about returning code different than 0. We can return it in the end if any of the examples fail to build.